### PR TITLE
Add tridiagonal solver to omp backend

### LINKED ISSUE #18: Add tridiagonal solver to omp backend
Use `test_omp_tridiag` as basis for tridiagonal solver

### PR DESCRIPTION
**Pull request by Nanoseb**
closes #18

### CODE REVIEW

File: tests/omp/test_omp_tridiag.f90 [hunk 1]
```diff
@@ -216,21 +214,21 @@ program test_omp_tridiag
...
 
    call check_error_norm(du, sin_0_2pi, n, n_glob, n_block, 1, norm_du)
-   if (nrank == 0) print*, 'error norm hyperviscous', norm_du
+   if (nrank == 0) print *, 'error norm hyperviscous', norm_du
 
    if (nrank == 0) then
       if (norm_du > tol) then
          allpass = .false.
-         write(stderr, '(a)') 'Check 2nd ders, hyperviscous, dir-neu... failed'
+        write (stderr, '(a)') 'Check 2nd ders, hyperviscous, dir-neu... failed'
```
Comment (semi-h) [inline tests/omp/test_omp_tridiag.f90:222]: seems like fprettify failed here for some reason, indentation level is wrong

File: tests/omp/test_omp_tridiag.f90 [hunk 2]
```diff
@@ -246,13 +244,13 @@ program test_omp_tridiag
 
    if (nrank == 0) then
       print'(a, f8.3, a)', 'Available BW:   ', deviceBW/2**30, &
-                           ' GiB/s (per NUMA zone on ARCHER2)'
+         ' GiB/s (per NUMA zone on ARCHER2)'
```
Comment (semi-h) [inline tests/omp/test_omp_tridiag.f90:247]: Is this also fprettify's auto fix?

File: tests/omp/test_omp_tridiag.f90 (hunk 1 above)
Comment (Nanoseb) [reply tests/omp/test_omp_tridiag.f90:222]: indeed, gonna fix that

File: tests/omp/test_omp_tridiag.f90 (hunk 2 above)
Comment (Nanoseb) [reply tests/omp/test_omp_tridiag.f90:247]: yes, it is also fprettify here. Though I don't mind that one too much

Comment (semi-h) [reply tests/omp/test_omp_tridiag.f90:247]: It looks okay. But if you're bothered with the whitespace between `print` and `*` you can add
`whitespace-print = false`
to the `.fprettify.ini` file if you want.  But then I think you can't put a space even after comma, so it becomes `print*,'something'` if I remember right. There is really not enough control over minor fprettify settings unfortunately.

Comment (Nanoseb) [reply tests/omp/test_omp_tridiag.f90:247]: I actually prefer `print *` over `print*`, so I don't mind the current syntax. I think I'd rather have `print '(xyz)'` versus `print'(xyz)'`, but I can live with that. As long as it is somewhat consistent and I don't have to manually tweak it I am fine with most syntax rule.

File: tests/omp/test_omp_tridiag.f90 (hunk 1 above)
Comment (Nanoseb) [reply tests/omp/test_omp_tridiag.f90]: Fixed

**Pull request merged**